File: src/clewin/RayMarchRenderer.java
package clewin;

import java.io.IOException;

/**
 * (c) Clewin Griffith 2013
 */
public class RayMarchRenderer {

	private RayMarcher rayMarcher;
	private RayGenerator rayGenerator;
	private Camera cam;
	private Image focalPlane;
	

	public RayMarchRenderer(RayMarcher rayMarcher, RayGenerator rayGenerator,
			Camera cam, Image focalPlane) {
		this.rayMarcher = rayMarcher;
		this.rayGenerator = rayGenerator;
		this.cam = cam;
		this.focalPlane = focalPlane;
	}
	
	public void render() throws IOException {
		Iterable<Ray> rays = rayGenerator.generateRays(cam, focalPlane);
		for(Ray ray : rays) {
			double val = rayMarcher.march(ray);
			focalPlane.write(ray.px, focalPlane.yres-ray.py-1, val);
		}
		focalPlane.save();
	}



	
}

File: src/clewin/RayMarchRenderer.scala
package clewin

import java.io.IOException

/**
 * (c) Clewin Griffith 2013
 */
class RayMarchRenderer(pixelColorCalculator: PixelColorCalculator, rayGenerator: RayGenerator, val cam: Camera, focalPlane: Image) {

  private val colorFilter = new Gamma()


  def render {
    val rays = rayGenerator.generateRays(cam, focalPlane)
    import scala.collection.JavaConversions._
    for (ray <- rays) {
      val pixelColor = pixelColorCalculator.calculatePixelColor(ray)
      val filteredColor = colorFilter.apply(pixelColor)
      //val `val`: Double = rayMarcher.march(ray)
      focalPlane.write(ray.px, focalPlane.yres - ray.py - 1, filteredColor)
    }
    focalPlane.save
  }



}